modules/REST/V2/Languages.php

```diff
@@ -0,0 +1,737 @@
+	 * @since 3.7
+	 *
+	 * @return void
+	 */
+	public function register_routes(): void {
+		register_rest_route(
+			$this->namespace,
+			"/{$this->rest_base}",
+			array(
+				array(
+					'methods'             => WP_REST_Server::READABLE,
+					'callback'            => array( $this, 'get_items' ),
+					'permission_callback' => '__return_true',
+				),
+				array(
+					'methods'             => WP_REST_Server::CREATABLE,
+					'callback'            => array( $this, 'create_item' ),
+					'permission_callback' => array( $this, 'create_item_permissions_check' ),
+					'args'                => $this->get_endpoint_args_for_item_schema( WP_REST_Server::CREATABLE ),
+				),
+				'schema' => array( $this, 'get_public_item_schema' ),
+			)
+		);
+
+		register_rest_route(
+			$this->namespace,
+			"/{$this->rest_base}/(?P<code>[a-z_-]+)",
+			array(
+				'args'   => array(
+					'code' => array(
+						'description' => __( 'Unique code for the language.', 'polylang' ),
+						'type'        => 'string',
+					),
+				),
+				array(
+					'methods'             => WP_REST_Server::READABLE,
+					'callback'            => array( $this, 'get_item' ),
+					'permission_callback' => '__return_true',
+					'args'                => array(
+						'context' => $this->get_context_param( array( 'default' => 'view' ) ),
+					),
+				),
+				array(
+					'methods'             => WP_REST_Server::EDITABLE,
+					'callback'            => array( $this, 'update_item' ),
+					'permission_callback' => array( $this, 'update_item_permissions_check' ),
+					'args'                => $this->get_endpoint_args_for_item_schema( WP_REST_Server::EDITABLE ),
+				),
+				array(
+					'methods'             => WP_REST_Server::DELETABLE,
+					'callback'            => array( $this, 'delete_item' ),
+					'permission_callback' => array( $this, 'delete_item_permissions_check' ),
+				),
+				'schema' => array( $this, 'get_public_item_schema' ),
+			)
+		);
+	}
+
+	/**
+	 * Retrieves a collection of languages.
+	 *
+	 * @since 3.7
+	 *
+	 * @param WP_REST_Request $request Full details about the request.
+	 * @return WP_REST_Response|WP_Error Response object on success, or WP_Error object on failure.
+	 *
+	 * @phpstan-template T of Fields
+	 * @phpstan-param WP_REST_Request<T> $request
+	 */
+	public function get_items( $request ) {
+		$languages = array();
+
+		foreach ( $this->model->languages_model->get_languages_list() as $language ) {
+			$languages[] = $this->prepare_item_for_response( $language, $request );
+		}
+
+		return rest_ensure_response( $languages );
+	}
+
+	/**
+	 * Creates one language from the collection.
+	 *
+	 * @since 3.7
+	 *
+	 * @param WP_REST_Request $request Full details about the request.
+	 * @return WP_REST_Response|WP_Error Response object on success, or WP_Error object on failure.
+	 *
+	 * @phpstan-template T of Fields
+	 * @phpstan-param WP_REST_Request<T> $request
+	 */
+	public function create_item( $request ) {
+		if ( ! empty( $request['id'] ) ) {
```

Comment:
Can this happen?
I would expect it to be WordPress business to filter out such requests.

Reply:
This can happen currently. What would you do to tell WP it's forbidden?

Reply:
To be discussed with the team.